packages/openapi-v3/src/decorators/request-body.option1.decorator.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright IBM Corp. 2018,2019. All Rights Reserved.
+// Node module: @loopback/openapi-v3
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import { MetadataInspector, ParameterDecoratorFactory } from '@loopback/context';
+import { JsonSchemaOptions } from '@loopback/repository-json-schema';
+import * as _ from 'lodash';
+import { inspect } from 'util';
+import { TS_TYPE_KEY } from '../controller-spec';
+import { resolveSchema } from '../generate-schema';
+import { OAI3Keys } from '../keys';
+import { RequestBodyObject, SchemaObject } from '../types';
+
+const debug = require('debug')('loopback:openapi3:metadata:requestbody');
+export const REQUEST_BODY_INDEX = 'x-parameter-index';
+
+export type SchemaOptions = JsonSchemaOptions & {
+  // If the schema options are provided, it implies
+  // the default paramter type need to be configured,
+  // so that `generateOpenAPISchema` can name the
+  // generated schema as `schemaName`
+  [TS_TYPE_KEY]?: Function,
+  isVisited?: boolean,
+}
```

Comment:
Maybe the following is better:
```ts
export interface SchemaOptions extends JsonSchemaOptions {
  tsType: {
    type: Function,
    exclude?: string[],
    optional?: string[],
  }
  isVisited?: boolean,
}
```

Reply:
I agree 👍

Reply:
Eventually I am able to use `JsonSchemaOptions` without creating a new interface/type. So removed it in the new PR https://github.com/strongloop/loopback-next/pull/3466 :)